src/openeo_aggregator/backend.py

```diff
@@ -748,16 +799,20 @@ def create_service(self, user_id: str, process_graph: dict, service_type: str, a
         """
         # TODO: configuration is not used. What to do with it?
 
+        # TODO: Strictly speaking it would be better to override _create_service instead of create_service
+        #       but for now we override create_service so we can keep the special case for the "sentinelhub" backend.
+
         # TODO: hardcoded/forced "SentinelHub only" support for now.
         #       Instead, properly determine backend based on service type?
         #       See https://github.com/Open-EO/openeo-aggregator/issues/78#issuecomment-1326180557
         #       and https://github.com/Open-EO/openeo-aggregator/issues/83
+        #       Should be able to remove this hardcoded workaround once issue #85 has been implemented.
         if "sentinelhub" in self._backends._backend_urls:
             backend_id = "sentinelhub"
         else:
-            backend_id = self._processing.get_backend_for_process_graph(
-                process_graph=process_graph, api_version=api_version
-            )
+            backend_id = self._find_backend_id_for_service_type(service_type)
```

Comment:
This implementation makes the hardcoded `if "sentinelhub" in ...`  hack unnecessary. 
Just replace the whole `if-else` with a simple `_find_backend_id_for_service_type` call

Reply:
Also remove the TODO from line 805 above

Reply:
Removed the hardcoded sentinelhub workaround and also the test that verifies it.